modules/porous_flow/include/materials/PorFlowMaterialMassFractionBuilder.h

```diff
@@ -0,0 +1,57 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+
+#ifndef PORFLOWMATERIALMASSFRACTIONBUILDER_H
+#define PORFLOWMATERIALMASSFRACTIONBUILDER_H
+
+#include "DerivativeMaterialInterface.h"
+#include "Material.h"
+
+#include "PorFlowVarNames.h"
+
+//Forward Declarations
+class PorFlowMaterialMassFractionBuilder;
+
+template<>
+InputParameters validParams<PorFlowMaterialMassFractionBuilder>();
+
+/**
+ * Material designed to form a std::vector<std::vector>
+ * of mass fractions from the individual mass fraction variables
+ */
+class PorFlowMaterialMassFractionBuilder : public DerivativeMaterialInterface<Material>
+{
+public:
+  PorFlowMaterialMassFractionBuilder(const InputParameters & parameters);
+
+protected:
+
+  unsigned int _num_phases;
+
+  unsigned int _num_components;
+
+  /// The variable names UserObject for the Porous-Flow variables
+  const PorFlowVarNames & _porflow_name_UO;
```

Comment:
Sooo.... there was that discussion about the `PorFlow` prefix... I know that it is quite bikeshed, but what are you going to go with now?